core/src/test/java/org/apache/calcite/test/enumerable/EnumerableCombineTest.java

```diff
@@ -0,0 +1,162 @@
+                  .aggregate(builder.groupKey(),
+                      builder.avg(builder.field("salary")).as("avg_salary"));
+
+              // Combine both queries
+              return builder.combine(2).build();
+            })
+        .returnsUnordered(
+            "QUERY=[{deptno=20, emp_count=1}, {deptno=10, emp_count=3}]",
+            "QUERY=[{avg_salary=9125.0}]");
+  }
+
+  /**
+   * Test that executes two queries with multiple columns each.
+   * Query 1: Select empid and name from employees in department 10
+   * Query 2: Select deptno and name from departments
+   */
+  @Test void testCombineMultipleColumns() {
+    tester(new HrSchema())
+        .withRel(
+            builder -> {
+              // Query 1: SELECT empid, name FROM emps WHERE deptno = 10
+              builder.scan("s", "emps")
+                  .filter(
+                      builder.equals(
+                          builder.field("deptno"),
+                          builder.literal(10)))
+                  .project(
+                      builder.field("empid"),
+                      builder.field("name"));
+
+              // Query 2: SELECT deptno, name FROM depts
+              builder.scan("s", "depts")
+                  .project(
+                      builder.field("deptno"),
+                      builder.field("name"));
+
+              // Combine both queries
+              return builder.combine(2).build();
+            })
+        .returnsUnordered(
+            "QUERY=[{empid=100, name=Bill}, {empid=150, name=Sebastian}, {empid=110, name=Theodore}]",
+            "QUERY=[{deptno=10, name=Sales}, {deptno=30, name=Marketing}, {deptno=40, name=HR}]");
+  }
+
+  /**
+   * Test that executes two queries returning different numbers of columns.
+   * Query 1: Select name (1 column)
+   * Query 2: Select empid, name, deptno (3 columns)
+   */
+  @Test void testCombineDifferentColumnCounts() {
+    tester(new HrSchema())
+        .withRel(
+            builder -> {
+              // Query 1: SELECT name FROM depts (1 column)
+              builder.scan("s", "depts")
+                  .project(builder.field("name"));
+
+              // Query 2: SELECT empid, name, deptno FROM emps WHERE deptno = 10 (3 columns)
+              builder.scan("s", "emps")
+                  .filter(
+                      builder.equals(
+                          builder.field("deptno"),
+                          builder.literal(10)))
+                  .project(
+                      builder.field("empid"),
+                      builder.field("name"),
+                      builder.field("deptno"));
+
+              // Combine both queries
+              return builder.combine(2).build();
+            })
+        .returnsUnordered(
+            "QUERY=[{name=Sales}, {name=Marketing}, {name=HR}]",
+            "QUERY=[{empid=100, name=Bill, deptno=10}, {empid=150, name=Sebastian, deptno=10}, "
+                + "{empid=110, name=Theodore, deptno=10}]");
```

Comment:
This is great for me. For the result set name QUERY, can we keep the default naming convention with auto-incrementing numbers like QUERY_0 and QUERY_1? By the way, just to add, for future commits, could we avoid using push -f? It makes it a bit difficult to compare the changes with the previous version.